.circleci/config.yml

```
@@ -107,3 +108,19 @@ jobs:
           name: Run system tests
           command: |
             COVERAGE=false bundle exec rspec spec/system/
+
+
+workflows:
+  test-build-and-publish-image:
+    jobs:
+      - run-tests
+      - wait-for-approval:
+          type: approval
+      - docker/publish:
+          cache_from: $CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME:latest
+          image: $CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME
+          tag: latest
+          update-description: false
+          requires:
+            - wait-for-approval
+            - run-tests
```

Is `wait-for-approval` needed here? Unless there's a strong reason for it, maybe we can publish on any merge to `main`. There's a `filters` key that allows us to run a workflow only on a particular branch:

```suggestion
          requires:
            - run-tests
          filters:
            branches:
              only: main
```

---

No, it wasn't needed - I thought you might want to control when the build/publish happens. It only triggering on `main` makes sense to me!